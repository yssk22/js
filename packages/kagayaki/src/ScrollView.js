/* @flow &*/
import React, { type Node } from 'react';
import { ScrollView, RefreshControl } from 'react-native';

type Props<T> = {
  refreshing?: boolean,
  data: Array<T>,
  onRefresh?: () => void,
  childComponentFunc: (v: T) => Node
};

const ScrollViewList = function<T>(props: Props<T>) {
  const data = props.data || [];
  const func = props.childComponentFunc;
  const refreshing = props.refreshing || false;
  return (
    <ScrollView
      refreshControl={<RefreshControl refreshing={refreshing} onRefresh={props.onRefresh} />}
    >
      {data.map(func)}
    </ScrollView>
  );
};

export default ScrollViewList;
