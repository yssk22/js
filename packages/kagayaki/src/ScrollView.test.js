/* @flow */
import React from 'react';
import { Text } from 'react-native';
import ScrollView from './ScrollView';
import renderer from 'react-test-renderer';

describe('ScrollView', () => {
  test('render', () => {
    const component = renderer.create(
      <ScrollView
        data={[{ a: 1 }]}
        childComponentFunc={(v, idx) => {
          return <Text key={idx}>{v.a}</Text>;
        }}
      />
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
