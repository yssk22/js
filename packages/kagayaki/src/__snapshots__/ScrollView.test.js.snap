// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScrollView render 1`] = `
<RCTScrollView
  refreshControl={
    <RefreshControlMock
      refreshing={false}
    />
  }
>
  <RCTRefreshControl />
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    >
      1
    </Text>
  </View>
</RCTScrollView>
`;
